registry-contract/ts/index.ts

```diff
@@ -1,65 +1,33 @@
-import * as wallet from 'ethereumjs-wallet'
-import Transaction from 'ethereumjs-tx'
+import { Contract, ethers } from "ethers";
+import { BaseProvider } from "ethers/providers";
 
 const RegistryContract = require('../build/contracts/Registry.json')
-const Web3 = require('web3')
 
-export default class EthereumResolver{
-  private web3: any
-  private indexContract: any
-  private contractAddress: string
+export default class EthereumResolver {
+  private readonly provider: BaseProvider
+  private contract: Contract
 
   constructor(address: string, providerUri: string) {
-    const provider =  new Web3.providers.HttpProvider(providerUri)
-    this.web3 = new Web3(provider)
-    this.contractAddress = address
-    this.indexContract = new this.web3.eth.Contract(RegistryContract.abi, address)
+    this.provider = new ethers.providers.JsonRpcProvider(providerUri)
+    this.contract = new ethers.Contract(address, RegistryContract.abi, this.provider)
   }
-  
-  resolveDID(did: string): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const keyHash = this._stripMethodPrefix(did)
-      this.indexContract.methods.getRecord(keyHash).call((error, result) => {
-        if (error) {
-          return reject(error)
-        }
-        return resolve(result)
-      })
-    })
-  }
-
-  updateDIDRecord(ethereumKey: any, did: string, newHash: string): Promise<string> {
-    const gasLimit = 250000
-    const gasPrice = 20e9
-
-    const w = wallet.fromPrivateKey(ethereumKey)
-    const address = w.getAddress().toString('hex')
-    const keyHash = this._stripMethodPrefix(did)
-
-    const callData = this.indexContract.methods.setRecord(keyHash, newHash)
-      .encodeABI()
 
-    return this.web3.eth.getTransactionCount(address).then(nonce => {
-      const tx = new Transaction({
-        nonce: nonce,
-        gasLimit,
-        gasPrice,
-        data: callData,
-        to: this.contractAddress
-      })
-
-      tx.sign(ethereumKey)
-      const serializedTx = tx.serialize()
-
-      return new Promise((resolve, reject) => {
-        this.web3.eth.sendSignedTransaction(`0x${serializedTx.toString('hex')}`)
-        .on('confirmation', () => resolve())
-        .on('error', (err) => reject(err))
-      })
-    })
+  async resolveDID(did: string): Promise<string> {
+    const keyHash = stripMethodPrefix(did)
+    return await this.contract.getRecord(keyHash)
   }
 
-  private _stripMethodPrefix(did: string): string {
-    return `0x${did.substring(did.lastIndexOf(':') + 1)}`
+  async updateDIDRecord(ethereumKey: any, did: string, newHash: string): Promise<string> {
+    const keyHash = stripMethodPrefix(did)
+
+    const wallet = new ethers.Wallet(ethereumKey, this.provider)
+    const signer = this.contract.connect(wallet)
+    const tx = await signer.setRecord(keyHash, newHash)
+    await tx.wait()
+    return ""
```

Comment:
The previous implementation had an empty return value as well, so this maintains the API, but we should make a note to make this more useful somehow instead of an empty string promise in future if we ever consider changing the API

Reply:
Yes this is just to maintain the API. #6 is fixing that